modules/git/commit_info.go

```diff
@@ -5,325 +5,233 @@
 package git
 
 import (
-	"bufio"
-	"context"
-	"fmt"
-	"os/exec"
-	"path"
-	"runtime"
-	"strconv"
-	"strings"
-	"sync"
-	"time"
+	"github.com/emirpasic/gods/trees/binaryheap"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
 )
 
-const (
-	// parameters for searching for commit infos. If the untargeted search has
-	// not found any entries in the past 5 commits, and 12 or fewer entries
-	// remain, then we'll just let the targeted-searching threads finish off,
-	// and stop the untargeted search to not interfere.
-	deferToTargetedSearchColdStreak          = 5
-	deferToTargetedSearchNumRemainingEntries = 12
-)
+// GetCommitsInfo gets information of all commits that are corresponding to these entries
+func (tes Entries) GetCommitsInfo(commit *Commit, treePath string, cache LastCommitCache) ([][]interface{}, *Commit, error) {
```

Comment:
Could you give some performance tests on this change?

Reply:
If you want just the numbers then they are already in the linked issue (#491). On our repository (lot of files of varying age, around 6Gb of data, 11 years of history) this is enough to go from loading the listing page in a minute to something like 1-4 seconds depending on file system cache and some other factors.